### PR TITLE
llms/googleai: batch embedding calls

### PR DESCRIPTION
**Pull request by johanbrandhorst**
The new Batch API significantly speeds up embedding

### PR Checklist

- [ ] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [ ] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [ ] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [ ] Describes the source of new concepts.
- [ ] References existing implementations as appropriate.
- [ ] Contains test coverage for new functions.
- [ ] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Comment by tmc** [comment]
So good to see you contributing! can't wait to get this in!.

File: llms/googleai/embeddings.go [hunk 1]
```diff
@@ -11,12 +12,20 @@ func (g *GoogleAI) CreateEmbedding(ctx context.Context, texts []string) ([][]flo
...
+		if (i > 0 && (i+1)%100 == 0) || i == len(texts)-1 {
+			embeddings, err := em.BatchEmbedContents(ctx, batch)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create embeddings: %w", err)
+			}
+			for _, e := range embeddings.Embeddings {
+				results = append(results, e.Values)
+			}
+			batch = em.NewBatch()
 		}
```
Comment (johanbrandhorst) [inline llms/googleai/embeddings.go:28]: The batching API allows 100 documents per batch, this automatically sends a request every 100 texts. I can add a comment if you prefer?

Comment (johanbrandhorst) [reply llms/googleai/embeddings.go:31]: Added a comment.

**Review by tmc — APPROVED** [review]
Certainly open to ideas on improving the testing strategy for this, putting this in now but do consider a test as you see fit.

**Comment by eliben** [comment]
Also happy to see you contribute here, @johanbrandhorst  :-)

Re tests, the `googleai` provider is relatively well-tested with live tests vs. the Gemini API (https://github.com/tmc/langchaingo/blob/main/llms/googleai/shared_test/shared_test.go)

For this test specifically, testing batching vs. no batching is a bit tricky and we don't really do mock testing with fake HTTP backends for now. That said, if the shared live tests pass and you observe the performance improvement, I believe this is good enough.

**Pull request merged**